lib/chromic_pdf/api/pdfa_options.ex

```diff
@@ -0,0 +1,22 @@
+defmodule ChromicPDF.PDFAOptions do
+  @moduledoc false
+
+  def feed_ghostscript_file_into_output(pdfa_path, opts) do
+    case Keyword.get(opts, :output) do
+      path when is_binary(path) ->
+        File.cp!(pdfa_path, path)
+        :ok
+
+      fun when is_function(fun, 1) ->
+        {:ok, fun.(pdfa_path)}
+
+      nil ->
+        data =
+          pdfa_path
+          |> File.read!()
+          |> Base.encode64()
+
+        {:ok, data}
+    end
+  end
```

Comment:
Reading this I was close to thinking if one could refactor the option handling to feel like a middleware stack, where each middleware would be handled on the way there and back to process specific options. 
Like this one could separate all this concerns into very small modules. 
But just a thought on the side, no action needed :)